src/components/action-popover/action-popover-menu/action-popover-menu.component.tsx

```diff
@@ -108,54 +115,114 @@ const ActionPopoverMenu = React.forwardRef<
       });
     }, [children]);
 
+    const isItemDisabled = useCallback(
+      (value: number) => {
+        const item = items[value];
+        // The invariant will be triggered before this else path can be explored, hence the ignore else.
+        // istanbul ignore else
+        return React.isValidElement(item) && item.props.disabled;
+      },
+      [items]
+    );
+
+    const findFirstFocusableItemIndex = () => {
```

Comment:
Suggestion: you should be able to use the existing JS array utils here, also I think we should avoid running the effect on every render

```
   const firstFocusableItem = items.findIndex((_, index) => !isItemDisabled(index));
    const lastFocusableItem = items.findLastIndex((_, index) => !isItemDisabled(index));


    useEffect(() => {
      if (isOpen && firstFocusableItem !== -1)
        setFocusIndex(firstFocusableItem);
    }, [isOpen, firstFocusableItem, setFocusIndex]);

Reply:
Good suggestion. The idea behind running the `useEffect` on every render was purely in case the disabled items were to change. For example: 

First item in the list was disabled first time around of opening, so the second item is focused. User closes the menu, does _something_ in the app that now allows the first item to be enabled. The next time the menu opens the first item should now be focused.